src/resmom/start_exec.c

```diff
@@ -6226,6 +6235,39 @@ start_exec(job *pjob)
 		if ((pjob->ji_qs.ji_svrflags & JOB_SVFLG_CHKPT) ||
 			(pjob->ji_qs.ji_svrflags & JOB_SVFLG_ChkptMig)) {
 
+			/* NULL value passed to hook_input.vnl */
+			/* means to assign */
+			/* vnode list using pjob->ji_host[].   */
+			mom_hook_input_init(&hook_input);
+			hook_input.pjob = pjob;
+
+			mom_hook_output_init(&hook_output);
+			hook_output.reject_errcode = &hook_errcode;
+			hook_output.last_phook = &last_phook;
+			hook_output.fail_action = &hook_fail_action;
+
+			switch ((hook_rc=mom_process_hooks(HOOK_EVENT_EXECJOB_BEGIN,
```

Comment:
Please double check to make sure when a job is restarted from checkpoint, only 1 instance of EXECJOB_BEGIN hook is executed. Since this is coming from start_exec(), which also happens on a regular job without checkpoint, want to make sure that  2 EXECJOB_BEGIN hooks don't execute. Just go back to your test logs to verify...

Reply:
This part of code is wrapped by an if block that checks for `JOB_SVFLG_CHKPT` or `JOB_SVFLG_ChkptMig` flag of the job. So this `mom_proces_hooks` is only executed for restarted jobs, and will not run on a regular job. I have verified this in the test log too.